sourcegraph/sourcegraph.proto

```diff
@@ -142,7 +146,9 @@ message Repo {
 	// Fork is whether this repository is a fork.
 	bool fork = 12;
 
-	// Mirror is whether this repository is a mirror.
+	// Mirror indicates whether this repo's canonical location is on
+	// another server. Mirror repos track their upstream and are not
+	// eligible for discovery on this server.
 	bool mirror = 13;
```

Comment:
You wouldn't really ever want to do it, but there would be no technological limitation preventing you from doing it.

Mirror just means automatically fetch/update this repo when the upstream changes, and show a little tag in the UI saying "mirror."





> On Jun 21, 2015, at 14:43, Beyang Liu <notifications@github.com> wrote:
> 
> In sourcegraph/sourcegraph.proto:
> 
> > @@ -142,7 +146,9 @@ message Repo {
> >  	// Fork is whether this repository is a fork.
> >  	bool fork = 12;
> >  
> > -	// Mirror is whether this repository is a mirror.
> > +	// Mirror indicates whether this repo's canonical location is on
> > +	// another server. Mirror repos track their upstream and are not
> > +	// eligible for discovery on this server.
> >  	bool mirror = 13;
> Why would you ever want to mirror something hosted on another Sourcegraph instance? In that case, shouldn't you just fork that repository to your own local instance? The relationship of mirrored vs non-mirrored federated vs non-mirrored non-federated is pretty complex and confusing. It'd be better to enforce a simple "right" way of doing things, even if it means some edge cases aren't covered IMO. If people really want to copy a repository, they can always fork it, right?
> 
> —
> Reply to this email directly or view it on GitHub.
>